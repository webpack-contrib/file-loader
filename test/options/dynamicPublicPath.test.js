/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('dynamicPublicPath', () => {
    test('{Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            dynamicPublicPath(url) {
              return `test/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });
  });
});
