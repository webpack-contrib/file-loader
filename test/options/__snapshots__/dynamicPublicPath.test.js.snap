// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options dynamicPublicPath {Function} 1`] = `"throw new Error(\\"Module parse failed: /Users/anvar/projects/quanttech/file-loader/src/index.js??ref--0!/Users/anvar/projects/quanttech/file-loader/test/fixtures/file.png Identifier directly after number (1:23)\\\\nYou may need an appropriate loader to handle this file type.\\\\n| module.exports = test/9c87cbf3ba33126ffd25ae7f2f6bbafb.png;\\");"`;
